File: src/main/java/de/steklopod/sec06/Lec07FluxInterval.java
package de.steklopod.sec06;

import de.steklopod.utils.Util;
import reactor.core.publisher.Flux;

import java.time.Duration;

public class Lec07FluxInterval {

    public static void main(String[] args) {

        Flux.range(1, 10)
                .delayElements(Duration.ofSeconds(1))
                .log()
                .subscribe(Util.subscriber());


        Util.sleepSeconds(60);
    }

}

File: src/main/kotlin/de/steklopod/sec06/Lec07FluxInterval.kt
package de.steklopod.sec06

import de.steklopod.utils.Util.sleepSeconds
import de.steklopod.utils.Util.subscriber
import reactor.core.publisher.Flux
import java.time.Duration

object Lec07FluxInterval {
    @JvmStatic
    fun main(args: Array<String>) {
        Flux.range(1, 10)
            .delayElements(Duration.ofSeconds(1))
            .log()
            .subscribe(subscriber())

        sleepSeconds(60)
    }
}
